keps/sig-scheduling/20191011-graduate-schedule-daemonset-pods-to-ga.md

```diff
@@ -0,0 +1,126 @@
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+ScheduleDaemonSetPods has been created in the past to schedule DaemonSet Pods by the default scheduler. We wish to graduate ScheduleDaemonSetPods feature to make scheduling decisions only in the scheduler and remove the scheduling related code in the DaemonSetController.
+
+## Motivation
+
+ScheduleDaemonSetPods has been beta'ed in 1.12.
+
+### Goals
+
++ Plan to promote ScheduleDaemonSetPods to the stable version.
++ Remove scheduling related codes in DaemonSetController.
+
+### Non-Goals
+
++ Changing API field or meaning
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+ScheduleDaemonSetPods attaches node affinity to DaemonSet pods, which let the default kubernetes scheduler to schedule pods on specific nodes. The DaemonSet controller uses several scheduler predicates to calculate the nodes which need to schedule DaemonSet pods and create pods with specific NodeAffinity.
+
+### Risks and Mitigations
+
+The major concern for graduating ScheduleDaemonSetPods to the stable version could be the overhead to the scheduler and the startup time of daemons. After we graduate this feature, the scheduler would select nodes for all of the DaemonSet pods, which may cause a lot of pods with NodeAffinity to be processed.
+
+## Design Details
+
+### Test Plan
+
+#### Existing Tests
+
+ScheduleDaemonSetPods currently has multiple tests in various components that use the feature.
+
+#### Needed Tests
+
+As a non-optional feature, there should be a conformance test for ScheduleDaemonSetPods.
```

Comment:
hm... `ScheduleDaemonSetPods` is a kind of refactor, no changes to DeamonSet behaviour; so I think that's not necessary :)